chain/client/src/stateless_validation/chunk_validator/mod.rs

```diff
@@ -753,11 +752,10 @@ impl Client {
     /// you can use the `processing_done_tracker` argument (but it's optional, it's safe to pass None there).
     pub fn process_chunk_state_witness(
         &mut self,
-        encoded_witness: EncodedChunkStateWitness,
+        witness: ChunkStateWitness,
+        raw_witness_size: ChunkStateWitnessSize,
```

Comment:
I checked the use case of raw_witness_size and it's being used in two places
- Validation of too large witnesses in orphan witness pool [here](https://github.com/near/nearcore/blob/9db9c616540520d125729c3357970a08e4964b7d/chain/client/src/stateless_validation/chunk_validator/orphan_witness_handling.rs#L59-L69)
- Orphan witness metrics [here](https://github.com/near/nearcore/blob/9db9c616540520d125729c3357970a08e4964b7d/chain/client/src/stateless_validation/chunk_validator/orphan_witness_pool.rs#L46)

For the size check, we can easily just move it to the partial witness tracker module.

I'm wondering whether it makes sense to push the metrics elsewhere (example decode_state_witness fn) so that we can have a cleaner interface and skip on sending the raw sizes everywhere?

Reply:
I totally agree that passing `raw_witness_size` around sucks as it pollutes the interface.

I think the proper solution here would be to get rid of orphan pool and merge its functionality into partial witness tracker. That is something I want to do in the long term, so I suggest keeping this size ugliness for now.

Reply:
> For the size check, we can easily just move it to the partial witness tracker module.

The size check is meant only for orphan witnesses. It's dangerous to apply size limit to all witnesses, as we could end up in a state when the witness is too big and we can't move the chain forward.